### ISSUE TITLE
Doma APT module-info.java is missing `provides javax.annotation.processing.Processor`

### ISSUE THREAD

**Issue by agentgt** [domaframework/doma#983]
**Describe the bug**

1. Make a Maven project.
1. Make the project modular with `module-info.java`

```
mvn clean install

[ERROR] Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: org.seasar.doma.internal.apt.processor.DomainProcessor Unable to get public no-arg constructor
```

**Expected behavior**

For the annotation processors to be found

**Actual behavior**

```
mvn clean install

[ERROR] Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: org.seasar.doma.internal.apt.processor.DomainProcessor Unable to get public no-arg constructor
```

**Environment (please complete the following information):**
- Doma version or git rev: 2.54
- IDE version (ie. Eclipse or IntelliJ IDEA): N/A
- Output of `java -version`: 20
- Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.9.4
- Database version: N/A
- JDBC Driver version: N/A

**Additional context**

```java
module org.seasar.doma.processor {
  requires java.compiler;
  requires org.seasar.doma.core;
}
```

Should be:

```java
module org.seasar.doma.processor {
  requires java.compiler;
  requires org.seasar.doma.core;

  provides javax.annotation.processing.Processor with org.seasar.doma.internal.apt.processor.DomainProcessor,
org.seasar.doma.internal.apt.processor.DataTypeProcessor,
org.seasar.doma.internal.apt.processor.ExternalDomainProcessor,
org.seasar.doma.internal.apt.processor.DomainConvertersProcessor,
org.seasar.doma.internal.apt.processor.EmbeddableProcessor,
org.seasar.doma.internal.apt.processor.EntityProcessor,
org.seasar.doma.internal.apt.processor.DaoProcessor,
org.seasar.doma.internal.apt.processor.ScopeProcessor,
org.seasar.doma.internal.apt.processor.SingletonConfigProcessor;
```

**Comment by agentgt** [comment]
After testing it doesn't really matter. Ultimately because of the way this library is designed you have to use:

```xml
<annotationProcessorPaths>
....
</<annotationProcessorPaths>
```

instead of just

```xml
<dependencies>
<dependency> dom apt as optional and provided </dependency>
</dependencies>
```

Because `org.seasar.doma.processor` can only be put on the classpath as that is all both Maven and Gradle support for APT (javac actually supports APT with modulepath) doma.core needs to as well. What happens though is `doma.core` will be on the module path in a modular environment.

In my library https://github.com/jstachio/jstachio you do not have to use `annotationProcessorPaths` and let the compiler discover precisely because I make sure every dependencies is shaded in and never access the annotations directly.

I highly recommend you consider shading `doma.core` into the `doma.processor` to avoid issues you like this. Ideally the annotations should be in their own jar as well.

**Comment by nakamura-to** [comment]
Shadow jar support was addressed in #992
